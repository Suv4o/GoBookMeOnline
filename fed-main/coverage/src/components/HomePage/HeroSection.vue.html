
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/HomePage/HeroSection.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/components/HomePage</a> HeroSection.vue</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>102/102</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>102/102</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;script lang="ts"&gt;
export default {
  name: 'HeroSection',
}
&lt;/script&gt;
&nbsp;
&lt;script setup lang="ts"&gt;
import { Assertions } from '../../types/guards'
&nbsp;
function focusSearchInput() {
  const searchBar: HTMLElement | null = document.querySelector('#search-bar input')
&nbsp;
  Assertions.isHTMLElement(searchBar)
  searchBar.focus()
}
&lt;/script&gt;
&nbsp;
&lt;template&gt;
  &lt;div class="relative bg-gray-50 overflow-hidden"&gt;
    &lt;div class="hidden sm:block sm:absolute sm:inset-y-0 sm:h-full sm:w-full" aria-hidden="true"&gt;
      &lt;div class="relative h-full max-w-7xl mx-auto"&gt;
        &lt;svg
          class="absolute right-full transform translate-y-1/4 translate-x-1/4 lg:translate-x-1/2"
          width="404"
          height="784"
          fill="none"
          viewBox="0 0 404 784"
        &gt;
          &lt;defs&gt;
            &lt;pattern
              id="f210dbf6-a58d-4871-961e-36d5016a0f49"
              x="0"
              y="0"
              width="20"
              height="20"
              patternUnits="userSpaceOnUse"
            &gt;
              &lt;rect x="0" y="0" width="4" height="4" class="text-gray-200" fill="currentColor" /&gt;
            &lt;/pattern&gt;
          &lt;/defs&gt;
          &lt;rect width="404" height="784" fill="url(#f210dbf6-a58d-4871-961e-36d5016a0f49)" /&gt;
        &lt;/svg&gt;
        &lt;svg
          class="absolute left-full transform -translate-y-3/4 -translate-x-1/4 md:-translate-y-1/2 lg:-translate-x-1/2"
          width="404"
          height="784"
          fill="none"
          viewBox="0 0 404 784"
        &gt;
          &lt;defs&gt;
            &lt;pattern
              id="5d0dd344-b041-4d26-bec4-8d33ea57ec9b"
              x="0"
              y="0"
              width="20"
              height="20"
              patternUnits="userSpaceOnUse"
            &gt;
              &lt;rect x="0" y="0" width="4" height="4" class="text-gray-200" fill="currentColor" /&gt;
            &lt;/pattern&gt;
          &lt;/defs&gt;
          &lt;rect width="404" height="784" fill="url(#5d0dd344-b041-4d26-bec4-8d33ea57ec9b)" /&gt;
        &lt;/svg&gt;
      &lt;/div&gt;
    &lt;/div&gt;
&nbsp;
    &lt;div class="relative pt-16 pb-16 sm:pb-24"&gt;
      &lt;div class="mt-16 default-container sm:mt-24"&gt;
        &lt;div class="text-center"&gt;
          &lt;h1 class="text-4xl tracking-tight font-extrabold text-gray-900 sm:text-5xl md:text-6xl"&gt;
            &lt;span class="block xl:inline"&gt;Organised your next&lt;/span&gt;
            {{ ' ' }}
            &lt;span class="block text-teal-600 xl:inline"&gt;online booking&lt;/span&gt;
          &lt;/h1&gt;
          &lt;p class="mt-3 max-w-md mx-auto text-base text-gray-500 sm:text-lg md:mt-5 md:text-xl md:max-w-3xl"&gt;
            Anim aute id magna aliqua ad ad non deserunt sunt. Qui irure qui lorem cupidatat commodo. Elit sunt amet
            fugiat veniam occaecat fugiat aliqua.
          &lt;/p&gt;
          &lt;div class="mt-5 max-w-md mx-auto sm:flex sm:justify-center md:mt-8"&gt;
            &lt;div class="rounded-md shadow"&gt;
              &lt;button
                data-testid="Main Button"
                class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-white bg-teal-600 hover:bg-teal-700 md:py-4 md:text-lg md:px-10"
                @click="focusSearchInput"
              &gt;
                Find your booking
              &lt;/button&gt;
            &lt;/div&gt;
            &lt;div class="mt-3 rounded-md shadow sm:mt-0 sm:ml-3"&gt;
              &lt;a
                href="#"
                class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-teal-600 bg-white hover:bg-gray-50 md:py-4 md:text-lg md:px-10"
              &gt;
                I'm a provider
              &lt;/a&gt;
            &lt;/div&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-09-25T02:56:50.658Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    