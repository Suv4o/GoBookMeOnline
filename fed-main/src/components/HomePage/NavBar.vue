<script lang="ts">
export default {
  name: 'NavBar',
}
</script>
<script setup lang="ts">
import { Popover, PopoverButton, PopoverPanel } from '@headlessui/vue'
import { MenuIcon, XIcon } from '@heroicons/vue/outline'
</script>

<template>
  <Popover>
    <div class="max-w-7xl mx-auto px-4 sm:px-6">
      <nav class="relative flex items-center justify-between sm:h-10 md:justify-center" aria-label="Global">
        <div class="flex items-center flex-1 md:absolute md:inset-y-0 md:left-0">
          <div class="flex items-center justify-between w-full md:w-auto">
            <a href="#" class="text-gray-500 hover:text-gray-600">
              <span class="sr-only">Go Book Me Online</span>
              <div class="flex items-center">
                <img class="h-8 w-auto sm:h-10" src="../../assets/images/logo.svg" alt="" />
                <p class="sm:ml-4 ml-2 sm:text-lg font-semibold">GoBookMe.Today</p>
              </div>
            </a>
            <div class="-mr-2 flex items-center md:hidden">
              <PopoverButton
                class="bg-gray-50 rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-teal-500"
              >
                <span class="sr-only">Open main menu</span>
                <MenuIcon class="h-6 w-6" aria-hidden="true" />
              </PopoverButton>
            </div>
          </div>
        </div>
        <div class="hidden md:absolute md:flex md:items-center md:justify-end md:inset-y-0 md:right-0">
          <span class="inline-flex rounded-md shadow">
            <a
              href="#"
              class="inline-flex items-center px-4 py-2 border border-transparent text-base font-medium rounded-md text-teal-600 bg-white hover:bg-gray-50"
            >
              Sign in
            </a>
            <a
              href="#"
              class="inline-flex items-center px-4 py-2 ml-4 border border-transparent text-base font-medium rounded-md text-white bg-teal-600 hover:bg-teal-700"
            >
              Sign up
            </a>
          </span>
        </div>
      </nav>
    </div>

    <transition
      enter-active-class="duration-150 ease-out"
      enter-from-class="opacity-0 scale-95"
      enter-to-class="opacity-100 scale-100"
      leave-active-class="duration-100 ease-in"
      leave-from-class="opacity-100 scale-100"
      leave-to-class="opacity-0 scale-95"
    >
      <PopoverPanel focus class="absolute z-10 top-0 inset-x-0 p-2 transition transform origin-top-right md:hidden">
        <div class="rounded-lg shadow-md bg-white ring-1 ring-black ring-opacity-5 overflow-hidden">
          <div class="px-5 pt-4 mb-4 flex items-center justify-between">
            <div class="flex items-center">
              <img class="h-8 w-auto" src="../../assets/images/logo.svg" alt="" />
              <p class="ml-2 font-semibold">GoBookMe.Today</p>
            </div>
            <div class="-mr-2">
              <PopoverButton
                class="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-teal-500"
              >
                <span class="sr-only">Close menu</span>
                <XIcon class="h-6 w-6" aria-hidden="true" />
              </PopoverButton>
            </div>
          </div>
          <a href="#" class="block w-full px-5 py-3 text-center font-medium text-white bg-teal-600 hover:bg-teal-700">
            Sign up
          </a>
          <p class="my-4 text-center text-base font-medium text-gray-500">
            Existing customer?
            <a href="#" class="text-teal-600 hover:text-teal-500"> Sign in </a>
          </p>
        </div>
      </PopoverPanel>
    </transition>
  </Popover>
</template>
