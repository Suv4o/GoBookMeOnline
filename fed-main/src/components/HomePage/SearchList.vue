<script lang="ts">
export default {
  name: 'SearchList',
}
</script>

<script setup lang="ts">
import DefaultSearchList from '../Default/DefaultSearchList/DefaultSearchList.vue'
import DefaultPagination from '../Default/DefaultPagination/DefaultPagination.vue'
</script>

<template>
  <default-search-list title="Locations" type="location" />
  <default-search-list title="People" type="person" />
  <div class="mt-4">
    <default-pagination />
  </div>
</template>
