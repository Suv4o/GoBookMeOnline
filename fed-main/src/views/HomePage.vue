<script lang="ts">
export default {
  name: 'HomePage',
}
</script>

<script setup lang="ts">
import HeroSection from '../components/HomePage/HeroSection.vue'
import DefaultSearchBar from '../components/Default/DefaultSearchBar/DefaultSearchBar.vue'
import SearchList from '../components/HomePage/SearchList.vue'
import { useNotification } from '../utils/composables/notiofication'
import { NotificationTypes } from '../store/notification'
import { onBeforeMount } from 'vue'
import { useValidator } from '../utils/composables/validator'

onBeforeMount(() => {
  showNotificationForCreatedUser()
})

function showNotificationForCreatedUser() {
  const params = new URLSearchParams(window.location.search)

  if (params.has('successfully-created') && params.get('successfully-created') === 'true') {
    window.history.pushState({}, document.title, '/')
    useNotification({
      type: NotificationTypes.Success,
      title: 'Successfully created!',
      message: 'Your account has been created. Make your next booking now!',
    })
  }

  if (params.has('user')) {
    const email = params.get('user')

    const { validProps } = useValidator({
      email: email ? email : '',
    })

    if (validProps.email.valid) {
      console.log(email)
    }
  }
}
</script>

<template>
  <hero-section />
  <div class="relative">
    <div id="search-bar" class="relative w-full z-20">
      <default-search-bar />
    </div>
    <div class="relative pt-24 sm:pt-12 md:pt-14 w-full z-10">
      <search-list />
    </div>
  </div>
</template>
