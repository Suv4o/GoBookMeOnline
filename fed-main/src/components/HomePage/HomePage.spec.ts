import { render, cleanup, fireEvent } from '@testing-library/vue'
import { describe, it, expect, vi, beforeAll } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import HomePage from './HeroSection.vue'

const container = document.createElement('div')

describe('HomePage', async () => {
  beforeAll(() => {
    const input = document.createElement('input')
    container.setAttribute('id', 'search-bar')
    input.setAttribute('type', 'text')
    container.appendChild(input)
  })

  // Needs to be placed first in the test suite to work.
  // If we place last it will fail because the input is not in the DOM.
  it('snap shot matches', async () => {
    const wrapper = render(HomePage)
    expect(wrapper).toMatchSnapshot()
    cleanup()
  })

  it('render component correctly and click on the main button to focus', async () => {
    const wrapper = render(HomePage, {
      container: document.body.appendChild(container),
    })

    const mainButton = wrapper.getByTestId('Main Button')
    await fireEvent.click(mainButton)

    cleanup()
  })

  it('call focusSearchInput() function', async () => {
    const wrapper = shallowMount(HomePage, {
      container: document.body.appendChild(container),
    })

    const spyFocusSearchInput = vi.spyOn(wrapper.vm, 'focusSearchInput')
    await wrapper.vm.focusSearchInput()
    expect(spyFocusSearchInput).toHaveBeenCalled()
    expect(spyFocusSearchInput).toReturnWith(void 0)
    spyFocusSearchInput.mockReset()
  })
})
