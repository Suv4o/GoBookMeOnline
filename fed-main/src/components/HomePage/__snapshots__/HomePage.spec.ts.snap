// Vitest Snapshot v1

exports[`HomePage > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="relative bg-gray-50 overflow-hidden"
      >
        <div
          aria-hidden="true"
          class="hidden sm:block sm:absolute sm:inset-y-0 sm:h-full sm:w-full"
        >
          <div
            class="relative h-full max-w-7xl mx-auto"
          >
            <svg
              class="absolute right-full transform translate-y-1/4 translate-x-1/4 lg:translate-x-1/2"
              fill="none"
              height="784"
              viewBox="0 0 404 784"
              width="404"
            >
              <defs>
                <pattern
                  height="20"
                  id="f210dbf6-a58d-4871-961e-36d5016a0f49"
                  patternUnits="userSpaceOnUse"
                  width="20"
                  x="0"
                  y="0"
                >
                  <rect
                    class="text-gray-200"
                    fill="currentColor"
                    height="4"
                    width="4"
                    x="0"
                    y="0"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#f210dbf6-a58d-4871-961e-36d5016a0f49)"
                height="784"
                width="404"
              />
            </svg>
            <svg
              class="absolute left-full transform -translate-y-3/4 -translate-x-1/4 md:-translate-y-1/2 lg:-translate-x-1/2"
              fill="none"
              height="784"
              viewBox="0 0 404 784"
              width="404"
            >
              <defs>
                <pattern
                  height="20"
                  id="5d0dd344-b041-4d26-bec4-8d33ea57ec9b"
                  patternUnits="userSpaceOnUse"
                  width="20"
                  x="0"
                  y="0"
                >
                  <rect
                    class="text-gray-200"
                    fill="currentColor"
                    height="4"
                    width="4"
                    x="0"
                    y="0"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#5d0dd344-b041-4d26-bec4-8d33ea57ec9b)"
                height="784"
                width="404"
              />
            </svg>
          </div>
        </div>
        <div
          class="relative pt-16 pb-16 sm:pb-24"
        >
          <div
            class="mt-16 default-container sm:mt-24"
          >
            <div
              class="text-center"
            >
              <h1
                class="text-4xl tracking-tight font-extrabold text-gray-900 sm:text-5xl md:text-6xl"
              >
                <span
                  class="block xl:inline"
                >
                  Organised your next
                </span>
                   
                <span
                  class="block text-teal-600 xl:inline"
                >
                  online booking
                </span>
              </h1>
              <p
                class="mt-3 max-w-md mx-auto text-base text-gray-500 sm:text-lg md:mt-5 md:text-xl md:max-w-3xl"
              >
                 Anim aute id magna aliqua ad ad non deserunt sunt. Qui irure qui lorem cupidatat commodo. Elit sunt amet fugiat veniam occaecat fugiat aliqua. 
              </p>
              <div
                class="mt-5 max-w-md mx-auto sm:flex sm:justify-center md:mt-8"
              >
                <div
                  class="rounded-md shadow"
                >
                  <button
                    class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-white bg-teal-600 hover:bg-teal-700 md:py-4 md:text-lg md:px-10"
                    data-testid="Main Button"
                  >
                     Find your booking 
                  </button>
                </div>
                <div
                  class="mt-3 rounded-md shadow sm:mt-0 sm:ml-3"
                >
                  <a
                    class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-teal-600 bg-white hover:bg-gray-50 md:py-4 md:text-lg md:px-10"
                    href="#"
                  >
                     I'm a provider 
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative bg-gray-50 overflow-hidden"
    >
      <div
        aria-hidden="true"
        class="hidden sm:block sm:absolute sm:inset-y-0 sm:h-full sm:w-full"
      >
        <div
          class="relative h-full max-w-7xl mx-auto"
        >
          <svg
            class="absolute right-full transform translate-y-1/4 translate-x-1/4 lg:translate-x-1/2"
            fill="none"
            height="784"
            viewBox="0 0 404 784"
            width="404"
          >
            <defs>
              <pattern
                height="20"
                id="f210dbf6-a58d-4871-961e-36d5016a0f49"
                patternUnits="userSpaceOnUse"
                width="20"
                x="0"
                y="0"
              >
                <rect
                  class="text-gray-200"
                  fill="currentColor"
                  height="4"
                  width="4"
                  x="0"
                  y="0"
                />
              </pattern>
            </defs>
            <rect
              fill="url(#f210dbf6-a58d-4871-961e-36d5016a0f49)"
              height="784"
              width="404"
            />
          </svg>
          <svg
            class="absolute left-full transform -translate-y-3/4 -translate-x-1/4 md:-translate-y-1/2 lg:-translate-x-1/2"
            fill="none"
            height="784"
            viewBox="0 0 404 784"
            width="404"
          >
            <defs>
              <pattern
                height="20"
                id="5d0dd344-b041-4d26-bec4-8d33ea57ec9b"
                patternUnits="userSpaceOnUse"
                width="20"
                x="0"
                y="0"
              >
                <rect
                  class="text-gray-200"
                  fill="currentColor"
                  height="4"
                  width="4"
                  x="0"
                  y="0"
                />
              </pattern>
            </defs>
            <rect
              fill="url(#5d0dd344-b041-4d26-bec4-8d33ea57ec9b)"
              height="784"
              width="404"
            />
          </svg>
        </div>
      </div>
      <div
        class="relative pt-16 pb-16 sm:pb-24"
      >
        <div
          class="mt-16 default-container sm:mt-24"
        >
          <div
            class="text-center"
          >
            <h1
              class="text-4xl tracking-tight font-extrabold text-gray-900 sm:text-5xl md:text-6xl"
            >
              <span
                class="block xl:inline"
              >
                Organised your next
              </span>
                 
              <span
                class="block text-teal-600 xl:inline"
              >
                online booking
              </span>
            </h1>
            <p
              class="mt-3 max-w-md mx-auto text-base text-gray-500 sm:text-lg md:mt-5 md:text-xl md:max-w-3xl"
            >
               Anim aute id magna aliqua ad ad non deserunt sunt. Qui irure qui lorem cupidatat commodo. Elit sunt amet fugiat veniam occaecat fugiat aliqua. 
            </p>
            <div
              class="mt-5 max-w-md mx-auto sm:flex sm:justify-center md:mt-8"
            >
              <div
                class="rounded-md shadow"
              >
                <button
                  class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-white bg-teal-600 hover:bg-teal-700 md:py-4 md:text-lg md:px-10"
                  data-testid="Main Button"
                >
                   Find your booking 
                </button>
              </div>
              <div
                class="mt-3 rounded-md shadow sm:mt-0 sm:ml-3"
              >
                <a
                  class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-teal-600 bg-white hover:bg-gray-50 md:py-4 md:text-lg md:px-10"
                  href="#"
                >
                   I'm a provider 
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
